pkgs/by-name/bi/bitwarden-menu/package.nix

```diff
@@ -1,28 +1,34 @@
 {
   lib,
   python3Packages,
-  fetchPypi,
+  fetchFromGitHub,
+  dmenu,
 }:
 
 python3Packages.buildPythonApplication rec {
   pname = "bitwarden-menu";
-  version = "0.4.5";
+  version = "0.4.5-unstable-2025-08-17";
   pyproject = true;
 
-  src = fetchPypi {
-    pname = "bitwarden_menu";
-    inherit version;
-    hash = "sha256-vUlNqSVdGhfN5WjDjf1ub32Y2WoBndIdFzfCNwo5+Vg=";
+  # Workaround: strip the suffix so setuptools_scm sees a valid PEP 440 version
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = builtins.elemAt (builtins.split "-" version) 0;
+
+  src = fetchFromGitHub {
+    owner = "firecat53";
+    repo = "bitwarden-menu";
+    rev = "b7567137b8d24edc3b6914d7581df3ec06392ce0";
+    hash = "sha256-gXdq8ublHgqXMr5Up51tLIFcTkntz/a7otMJXeznNDU=";
   };
 
-  nativeBuildInputs = with python3Packages; [
-    hatch-vcs
-    hatchling
+  build-system = [
+    python3Packages.hatch-vcs
+    python3Packages.hatchling
   ];
 
-  propagatedBuildInputs = with python3Packages; [
-    pynput
-    xdg-base-dirs
+  dependencies = [
+    python3Packages.pynput
+    python3Packages.xdg-base-dirs
+    dmenu
```

Comment:
A few points:

I'm not super in tune with the Python packaging idioms, but to me `dependencies` should only be used for Python packages, and `dmenu` should be in the _actual_ `propagatedBuildInputs`. Please chime in if someone more knowledgeable sees this.

Secondly, this is probably more aptly done by appending `dmenu` to PATH in the wrapper, rather than propagating the package. Then `nix run` will work correctly.

Thirdly, I am ambivalent about providing `dmenu` as a fallback, or just relying on the user to have `rofi`/`dmenu` installed _as well_. It _does_ make e.g: `nix run` work even if it _isn't_ installed.